OpenJDK 21 support (#28518)
keycloak/keycloak: docs/documentation/release_notes/topics/25_0_0.adoc

Reviewer: With regards to docs, IMHO we should also update the Getting Started guides that instruct users to install Java 17.

The cookie `KC_AUTH_STATE` is removed and it is no longer set by the {project_name} server as this server no longer needs this cookie.

= Java 21 support

{project_name} now supports OpenJDK 21, as we want to stick to the latest LTS OpenJDK versions.

> = Java 17 is deprecated

Reviewer: ```suggestion
= Java 17 support is deprecated
```

> OpenJDK 17 is deprecated, and the support will be removed in the following releases in favor of OpenJDK 21.

Reviewer: ```suggestion
OpenJDK 17 support is deprecated in {project_name}, and will be removed in a following release in favor of OpenJDK 21.
```

> It is still possible to use it, but the test coverage might be decreased.

Reviewer 2: ```suggestion
```

Author: ok
